brickpi/brickpi_i2c_sensor.c

```diff
@@ -96,12 +96,27 @@ static int brickpi_i2c_sensor_send_command(void *context, u8 mode)
 	const struct nxt_i2c_sensor_mode_info *i2c_mode_info = data->info->i2c_mode_info;
 	int size = lego_sensor_get_raw_data_size(mode_info);
 
+	int err; /* temporary workaround for ms-absolute-imu (2 of 5) */
+	struct nxt_i2c_sensor_data workaround_sensor; /*workaround (3 of 5) */
+
 	/* set mode function also works for sending command */
-	return brickpi_in_port_set_i2c_mode(data->ldev,
+	err = brickpi_in_port_set_i2c_mode(data->ldev,
 					    i2c_cmd_info[mode].cmd_reg,
 					    i2c_cmd_info[mode].cmd_data,
 					    i2c_mode_info[mode].read_data_reg,
 					    size);
+
+	/* temporary workaround for ms-absolute-imu (4 of 5) */
+	if (err)
+		return err;
+
+	/* temporary workaround for ms-absolute-imu (5 of 5) */
+	if (data->info->ops && data->info->ops->send_cmd_post_cb) {
```

Comment:
any other sensor with non NULL ops wouldn't get through probe line:

``` 
if (sensor_info->ops && ldev->entry_id->driver_data != MS_ABSOLUTE_IMU) {
```

 but I agree that extra safety here would not hurt